collector/src/toolchain.rs

```diff
@@ -598,3 +598,58 @@ fn get_lib_dir_from_rustc(rustc: &Path) -> anyhow::Result<PathBuf> {
 
     Ok(Path::new(sysroot_path.as_ref().trim()).join("lib"))
 }
+
+#[cfg(test)]
+mod tests {
+    use rand::Rng;
+    use std::collections::HashMap;
+
+    use super::*;
+
+    #[test]
+    fn fill_libraries() {
+        let mut components = ToolchainComponents::default();
+        let mut libraries_to_read = vec!["librustc_driver.so", "libstd.so", "libtest.so"];
+        let llvms = vec![
+            "libLLVM-17-rust-1.76.0-stable.so",    // old LLVM
+            "libLLVM.so.18.1-rust-1.78.0-nightly", // new LLVM
+        ];
+        let mut libraries_path_map = HashMap::new();
+
+        // create a random case for testing LLVM
+        let mut rng = rand::thread_rng();
```

Comment:
This seems to be unnecessarily complicated and over-engineered :sweat_smile: And also having random/nondeterministic tests is not a good idea, when it's not needed.

Could you please simplify the test to this:
1) Create a directory with LLVM, rustc and libstd `.so` files
2) Call `fill_libraries` on the directory
3) Assert that the paths of the component attributes equals the created paths.

And then you can create two of these tests, one for the old and one for the new LLVM format.